### PR TITLE
feat: [FC-9] show typing when input text

### PR DESCRIPTION
**Pull request by huydosgtech**

### CODE REVIEW

**Comment by huydosgtech** [comment]
Image demo:

Screenshot 2024-06-17 at 1 20 07 PM

File: src/utilities/misc.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export const isOtherUserTyping = (
+  typingData: { [key: string]: boolean },
+  currentUserId: string
+): boolean => {
+  return Object.keys(typingData)
+    .filter((userId) => userId !== currentUserId)
```
Comment (loc-nguyenthien) [inline src/utilities/misc.ts:6]: not need

File: src/utilities/misc.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    .some((userId) => typingData[userId] === true);
```
Comment (loc-nguyenthien) [inline src/utilities/misc.ts:7]:
```suggestion
    .some((userId) => userId !== currentUserId && typingData[userId] === true);
```

**Review by loc-nguyenthien — CHANGES_REQUESTED** [review]

File: src/utilities/misc.ts (hunk 2 above)
Comment (huydosgtech) [reply src/utilities/misc.ts]: ok I updated. Please check again. Thanks

File: src/hooks.ts [hunk 3]
```diff
@@ -21,3 +22,59 @@ export const useChatSelector = <T>(
...
+    changeUserConversationTyping(false);
+    if (typingTimeoutRef.current) {
+      clearTimeout(typingTimeoutRef.current);
+      typingTimeoutRef.current = undefined;
+    }
+  }, [changeUserConversationTyping]);
+
+  const handleTextChange = useCallback(
+    (newText: string) => {
+      if (enableTyping) {
```
Comment (baonguyenhsts) [inline src/hooks.ts:59]: from 59 - 72
```suggestion
     if (!enableTyping) return;

    if (newText.trim().length > 0) {
      clearTimeout(typingTimeoutRef.current);
      typingTimeoutRef.current = setTimeout(stopTyping, typingTimeoutSeconds);
      startTyping();
      return;
    }
    stopTyping();
```

**Review by baonguyenhsts — CHANGES_REQUESTED** [review]

File: src/hooks.ts (hunk 3 above)
Comment (huydosgtech) [reply src/hooks.ts]: ok I updated it

File: src/utilities/misc.ts (hunk 1 above)
Comment (huydosgtech) [reply src/utilities/misc.ts]: ok I resolved

**Review by hoangSTS — APPROVED** [review]

**Comment by huydosgtech** [comment]
new UI
Simulator Screenshot - iPhone 15 - 2024-07-30 at 13 29 35

**Review by danhmaisgt — APPROVED** [review]

**Review by loc-nguyenthien — APPROVED** [review]

**Pull request merged**